exploration/0002-expression-attributes.md

```diff
@@ -38,18 +38,59 @@ At least the following expression attributes should be considered:
 
   - `fallback` — A value to use instead of the default fallback,
     should the expression's primary formatting fail in some way.
+
+    > Example that would format as "Hello world" if `$place` is not defined:
+    >
+    > ```
+    > {Hello {$place @fallback=world}}
+    > ```
+
   - `locale` — An override for the locale used to format the expression.
     Should be expressed as a non-empty sequence of BCP 47 language codes.
+
+    > Example embedding a French literal in an English message:
+    >
+    > ```
+    > {In French, "{|bonjour| @locale=fr}" is a greeting}
+    > ```
+
   - `dir` — An override for the LTR/RTL/auto directionality of the expression.
 
+    > Example explicitly isolating the directionality of a placeholder:
+    >
+    > ```
+    > {Welcome, {$username @dir=auto}}
```

Comment:
Same as above, why not encourage solving this via spans?

```
{Welcome, {+span dir=auto}{$username}{-span}.}
```

Reply:
In this case, where the thing affected is not a literal but actually a real placeholder, using a direct attribute is better I think.